pkg/database/upsertCluster_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package database
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stolostron/search-indexer/pkg/model"
+)
+
+var clusterProps map[string]interface{}
+var existingCluster map[string]interface{}
+
+func initializeVars() {
+	clusterProps = map[string]interface{}{
+		"apigroup":          "internal.open-cluster-management.io",
+		"consoleURL":        "",
+		"cpu":               0,
+		"created":           "0001-01-01T00:00:00Z",
+		"kind":              "Cluster",
+		"kubernetesVersion": "",
+		"memory":            0,
+		"name":              "name-foo",
+		"nodes":             0}
+	existingCluster = map[string]interface{}{"UID": "cluster__name-foo",
+		"Kind":       "Cluster",
+		"Properties": clusterProps}
+}
+func Test_UpsertCluster_NoUpdate(t *testing.T) {
+	initializeVars()
+	ExistingClustersCache = make(map[string]interface{})
+
+	// clusterResource := `{"UID":"cluster__name-foo", "Kind":"Cluster", "Properties":{"_clusterNamespace":"name-foo", "apigroup":"internal.open-cluster-management.io", "consoleURL":"", "cpu":0, "created":"0001-01-01T00:00:00Z", "kind":"Cluster", "kubernetesVersion":"", "memory":0, "name":"name-foo", "nodes":0}}`
+
+	// _ = json.Unmarshal([]byte(clusterResource), &existingCluster)
```

Comment:
Delete commented code. Same below.

The new format is easier to read! Can we use this format in `clusterSync_test.go`?

Reply:
Done